sycl/source/detail/usm/usm_impl.cpp

```diff
@@ -252,6 +265,10 @@ void *alignedAlloc(size_t Alignment, size_t Size, const context &Ctxt,
   PrepareNotify.scopedNotify(
       (uint16_t)xpti::trace_point_type_t::mem_alloc_begin);
 #endif
+  if (Kind == alloc::shared && !Dev.has(sycl::aspect::usm_shared_allocations)) {
+    throw sycl::exception(sycl::errc::feature_not_supported,
+                          "Device does not support USM shared allocations!");
+  }
```

Comment:
Can we move this to alignedAllocInternal? I think it's better to have both checks nearby.